staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -45,6 +50,194 @@ func TestDecode(t *testing.T) {
 		want          interface{}
 		assertOnError func(t *testing.T, e error)
 	}{
+		{
+			name: "reject duplicate keys (int)",
+			in:   hex("a201010102"), // {1:1, 1:2}
+			into: map[int64]interface{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if e.Index != 1 {
+					t.Errorf("expected %#v, got %#v", &cbor.DupMapKeyError{Index: 1}, e)
+				}
+			}),
+		},
+		{
+			name: "reject duplicate keys (text tring) (map)",
+			in:   hex("A2614101614102"), // {A:1, A:2}
+			into: map[string]interface{}{},
+			assertOnError: assertOnConcreteError(func(t *testing.T, e *cbor.DupMapKeyError) {
+				if e.Index != 1 {
+					t.Errorf("expected %#v, got %#v", &cbor.DupMapKeyError{Index: 1}, e)
+				}
+			}),
+		},
+		{
+			name:  "reject duplicate keys (text tring) (struct)",
+			in:    hex("A2614101614102"), // {A:1, A:2}
+			modes: []cbor.DecMode{modes.DecodeLax},
+			into:  struct{}{},
```

Comment:
This test case could cover both modes by giving the destination type a field that matches A (separately, it's still valuable to make sure we catch duplicate keys that don't match any field as this is doing).

```suggestion
			modes: []cbor.DecMode{modes.DecodeLax},
			into:  struct{A int `json:"A"`}{},
```